### DISCUSSION TITLE
backup of the closed key

### DISCUSSION THREAD

**Discussion by sidielnikov** [samuel-lucas6/Kryptor#36] [Ideas]
Hi samuel-lucas6, how do you like the idea to make a backup copy of a private key in an encrypted directory?

and adding BIP39 to reconcile a private key

1 generation of random bytes
2 GenerateKeyPairFromseed (1)
3 Print (BIP39 (1))

You can also screw the secret of the shamir. If the source code of algorithms are needed, contact

**Comment by samuel-lucas6** [comment]
Hi @sidielnikov, I'm not quite sure what you mean. You can back up the encrypted private key by either copying the file or copying the string contained inside the file. What would the advantage of having BIP39 private keys be?

**Reply by sidielnikov** [reply]
> Hi @sidielnikov, I'm not quite sure what you mean. You can back up the encrypted private key by either copying the file or copying the string contained inside the file. What would the advantage of having BIP39 private keys be?

Copy the encrypted private key file that is in .kryptor in an encrypted folder

The advantage is that if the key is lost, then you can reclaim it with a seprase

**Reply by samuel-lucas6** [reply]
> Copy the encrypted private key file that is in .kryptor in an encrypted folder

Why would you like to do this when it's already encrypted? Is it to remember which key was used to encrypt the directory?

> The advantage is that if the key is lost, then you can reclaim it with a seprase

But you need to remember the seed phrase still, which is not short, so most people would have to write that down anyway. At that point, you're doing exactly the same as backing up the private key, except the seed phrase isn't encrypted.

**Reply by sidielnikov** [reply]
> > Copy the encrypted private key file that is in .kryptor in an encrypted folder
> 
> Why would you like to do this when it's already encrypted? Is it to remember which key was used to encrypt the directory?
> 
> > The advantage is that if the key is lost, then you can reclaim it with a seprase
> 
> But you need to remember the seed phrase still, which is not short, so most people would have to write that down anyway. At that point, you're doing exactly the same as backing up the private key, except the seed phrase isn't encrypted.

yes, suddenly I lose the key file

**Reply by samuel-lucas6** [reply]
You've lost your private key file?

**Reply by sidielnikov** [reply]
> You've lost your private key file?

no, but this can happen from others

**Reply by samuel-lucas6** [reply]
Not if they follow the guidance [here](https://www.kryptor.co.uk/tutorial#generating-a-new-key-pair) and [here](https://www.kryptor.co.uk/best-practices#storing-your-public-and-private-keys) on the website. I'd also like to think most people capable of using a command line application are aware that important files should be backed up, and as I've already said, a seed phrase also needs to be backed up somewhere. However, perhaps a warning could be given in the `-g|--generator` output about backing up the private key file.

**Comment by sidielnikov** [comment]
I discovered the bug with repeated encryption of the names of the folders. When adding new files to the encrypted folder encrypted catalog names is encrypted again

**Reply by samuel-lucas6** [reply]
This is a bit off topic for this discussion. That shouldn't be possible since you shouldn't be able to encrypt an already encrypted directory.

**Reply by sidielnikov** [reply]
> This is a bit off topic for this discussion. That shouldn't be possible since you shouldn't be able to encrypt an already encrypted directory.

I mean the name of the catalog

**Reply by samuel-lucas6** [reply]
The name of the parent directory? I don't think that should be possible either.

**Reply by sidielnikov** [reply]
> The name of the parent directory? I don't think that should be possible either.

try to encrypt the directory then add files to it and encrypt again and see

**Reply by samuel-lucas6** [reply]
Nope, I still get an error. Please can you submit a [bug report](https://github.com/samuel-lucas6/Kryptor/issues/new/choose) issue to document this problem. Are you using the latest version?

**Reply by sidielnikov** [reply]
> Nope, I still get an error. Please can you submit a [bug report](https://github.com/samuel-lucas6/Kryptor/issues/new/choose) issue to document this problem. Are you using the latest version?

Yes, the latter
